runtime/doc/lua.txt

```diff
@@ -2970,6 +2979,30 @@ vim.fs.normalize({path}, {opts})                          *vim.fs.normalize()*
 
         vim.fs.normalize('$XDG_CONFIG_HOME/nvim/init.vim')
         -- '/Users/jdoe/.config/nvim/init.vim'
+
+        vim.fs.normalize('~/src/nvim/api/../tui/./tui.c')
+        -- '/home/jdoe/src/nvim/tui/tui.c'
```

Comment:
off topic, somewhat surprised that `~` is expanded without `expand_env=true` (which perhaps should be named `expand`)

Reply:
Fwiw, `~` is not expanded in Python's `os.path` either

Reply:
Not really; that's the core part of "normalization". `expand_env` is about `$HOME` and friends (only), and it makes sense that these are independent choices.

Reply:
My initial gut reaction was the same as Justin's (aka that `~` shouldn't be expanded if `expand_env` is false) although I can't with certainty say which is better/more consistent. Just wanted to share my 2 cents.

Reply:
> `expand_env` is about `$HOME` and friends (only)

That's an artificial/accidental constraint that we can choose to change. Or we could add a separate `expand` option in the worst case.

> that's the core part of "normalization"

No, I've made this clear in multiple discussions: normalization means "the result is deterministic". That is separate from "the result is different depending on the context".

With the current behavior, the caller has to care that `~/` will be expanded even if they don't want that. For example, saving a path to a config file you probably don't want `~/` to be expanded.